src/util/pluginError.ts

```diff
@@ -44,8 +44,14 @@ const ignoredPluginErrorProps = new Set([
   'showProperties',
   'showStack',
   'stack',
+  'target',
 ]);
 
+function hasDiagnosticTarget<T extends PluginError>(err: T):
+  err is T & { target: DiagnosticTarget } {
```

Comment:
The indentation is a little weird, having the return type and the body at the same indent level with nothing to break it up. Try this:
```suggestion
  err is T & { target: DiagnosticTarget }
{
```